src/wasm-interpreter.h

```diff
@@ -2240,6 +2463,22 @@ class ModuleInstance
     : ModuleInstanceBase(wasm, externalInterface) {}
 };
 
+// Expression runner exposed by the C-API
+class CExpressionRunner final : public ExpressionRunner<CExpressionRunner> {
```

Comment:
Is there any reason not to define this subclass in binaryen-c.cpp and the PrecomputingExpressionRunner in Precompute.cpp?

Reply:
Defining it in `binaryen-c.cpp` yielded an error that the size of `struct NonconstantException` is `0 in C and 1 in C++`. Going to move that struct as per your suggestion, so might well be that this isn't an issue anymore.

Reply:
Yep, that worked